lawnchair/src/app/lawnchair/ui/preferences/components/IconShapePreference.kt

```diff
@@ -82,20 +109,101 @@ fun IconShapePreference(
                     enabled = item.enabled,
                     title = { Text(item.label()) },
                     modifier = Modifier.clickable(item.enabled) {
-                        adapter.onChange(newValue = item.value)
+                        iconShapeAdapter.onChange(newValue = item.value)
                     },
                     startWidget = {
                         RadioButton(
-                            selected = item.value == adapter.state.value,
+                            selected = item.value == iconShapeAdapter.state.value,
                             onClick = null,
                             enabled = item.enabled,
                         )
                     },
                     endWidget = {
                         IconShapePreview(iconShape = item.value)
-                    }
+                    },
                 )
             }
+            CustomIconShapePreference(
+                iconShapeAdapter = iconShapeAdapter,
+            )
+            ModifyCustomIconShapePreference(
+                customIconShape = customIconShape.value,
+            )
+        }
```

Comment:
I'd recommend separating this from the "presets" area.

If you will soon add "saving icon shapes", maybe consider adding a "custom" label:


```suggestion
        }
            
        PreferenceGroup (
            heading = stringResource(id = R.string.custom),
        ) {
            CustomIconShapePreference(
                iconShapeAdapter = iconShapeAdapter,
            )
            ModifyCustomIconShapePreference(
                customIconShape = customIconShape.value,
            )
        }
```

Reply:
Was a bit unsure if it would look better, but after testing it I actually like it much better than the previous style.

![Peek 2022-10-12 16-15](https://user-images.githubusercontent.com/41836211/195345919-a8b07f72-f90d-4fd5-a259-828fa1896024.gif)

Reply:
Yeah, I think it looks great. You can add it if you want.